### PR TITLE
build-support/php: `phpDrv` in `nativeBuildInputs`

### LINKED ISSUE #261500: php: `buildComposerProject` using wrong PHP package?
### Describe the bug
I might be misunderstanding, but I believe that there's a bug in `php.buildComposerProject` (or below), since I don't get the `php` I expect in my build script.

### Steps To Reproduce
Try to build the following:

```nix
{ lib, php, runCommand, which }: let
  phpWithExtensions = php.withExtensions ({ enabled, all }: enabled ++ [all.redis]);
in
phpWithExtensions.buildComposerProject (finalAttrs: {
  pname = "test";
  version = "0.0.1";
  src = runCommand "test" {} ''
    mkdir -p $out
    echo """
    {
      \"name\": \"test/test\",
      \"description\": \"test\",
      \"license\": \"GPL-3.0-or-later\"
    }
    """ > $out/composer.json
    echo """
    {
      \"_readme\": [
        \"This file locks the dependencies of your project to a known state\",
        \"Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies\",
        \"This file is @generated automatically\"
      ],
      \"content-hash\": \"2f3a1ed4816797820ab9fdbc67c25cc8\",
      \"packages\": [],
      \"packages-dev\": [],
      \"aliases\": [],
      \"minimum-stability\": \"stable\",
      \"stability-flags\": [],
      \"prefer-stable\": false,
      \"prefer-lowest\": false,
      \"platform\": [],
      \"platform-dev\": [],
      \"plugin-api-version\": \"2.3.0\"
    }
    """ > $out/composer.lock
  '';
  vendorHash = "sha256-xuPbVsNwIpVVsXw++0bvevNGwDYZbFvKQNvUauN+Zx8=";

  # Un-commenting this does not help.
  #php = phpWithExtensions;

  nativeBuildInputs = [ which ];
  buildPhase = ''
    # The following two lines are different?!
    echo $(readlink -f $(which php))
    echo ${lib.getExe phpWithExtensions}

    # Missing extension is how I discovered this bug.
    php --ri redis
  '';
})
```

### Expected behavior
`php` inside the derivation should be the same as `phpWithExtensions`, and `php --ri redis` should succeed.

### Notify maintainers

@drupol

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.44, NixOS, 23.05 (Stoat)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(root): `"nixos-20.09.2674.a058d005b3c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by lorenzleutgeb**
Fixes NixOS/nixpkgs#261500

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md)) (does not apply)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/php/build-composer-project.nix [hunk 1]
```diff
@@ -17,11 +17,11 @@ let
       nativeBuildInputs = (previousAttrs.nativeBuildInputs or [ ]) ++ [
         composer
         composer-local-repo-plugin
+        phpDrv
         phpDrv.composerHooks.composerInstallHook
       ];
 
       buildInputs = (previousAttrs.buildInputs or [ ]) ++ [
-        phpDrv
```
Comment (drupol) [inline pkgs/build-support/php/build-composer-project.nix:24]: I would also change the previous line to:

```nix
buildInputs = previousAttrs.buildInputs or [ ];
```

**Review by drupol — CHANGES_REQUESTED** [review]
Thank you for finding the issue and providing the fix! Very much appreciated :)

A small nit and I think it's good to go.

**Comment by drupol** [comment]
@GrahamcOfBorg build php.packages.composer

**Comment by drupol** [comment]
By the way, how about removing the `php` derivation from the hooks as well ?

**Comment by lorenzleutgeb** [comment]
On the second thought, I now think that `phpDrv` should be included in *both* `nativeBuildInputs` and `buildInputs`.

**Comment by drupol** [comment]
@GrahamcOfBorg build php.packages.composer

**Review by drupol — APPROVED** [review]
This is definitely good for me!

However, I'll wait for @etu's review since this is touching something critical parts and he might have something to say.

**Review by etu — APPROVED** [review]

**Comment by drupol** [comment]
Great, thank you both !

**Pull request merged**